src/core/manager.c

```diff
@@ -3491,19 +3467,15 @@ int manager_reload(Manager *m) {
         if (!fds)
                 return log_oom();
 
-        /* We are officially in reload mode from here on */
-        m->n_reloading++;
+        /* We are officially in reload mode from here on. */
+        _cleanup_(manager_reloading_stopp) Manager *reloading = manager_reloading_start(m);
 
         r = manager_serialize(m, f, fds, false);
-        if (r < 0) {
-                log_error_errno(r, "Failed to serialize manager: %m");
-                goto fail;
-        }
+        if (r < 0)
+                return log_error_errno(r, "Failed to serialize manager: %m");
 
-        if (fseeko(f, 0, SEEK_SET) < 0) {
-                r = log_error_errno(errno, "Failed to seek to beginning of serialization: %m");
-                goto fail;
-        }
+        if (fseeko(f, 0, SEEK_SET) < 0)
+                return log_error_errno(errno, "Failed to seek to beginning of serialization: %m");
 
         /* 💀 This is the point of no return, from here on there is no way back. 💀 */
```

Comment:
It may be better to insert `reloading = NULL;` here.

Reply:
@yuwata: thanks!

Updated with the suggested changes, no other changes.